files/image_config/dhcp_dos_logger/dhcp_dos_logger.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+import re
+import os
+import subprocess
+import time
+from sonic_py_common.logger import Logger
+from swsscommon.swsscommon import ConfigDBConnector
+
+SYSLOG_IDENTIFIER = os.path.basename(__file__)
+
+# Global logger instance
+logger = Logger(SYSLOG_IDENTIFIER)
+logger.log_info("Starting DHCP DoS logger...")
+
+# Connect to config db
+config_db = ConfigDBConnector()
+config_db.connect()
+
+# Initialize
+drop_pkts = {}
+
+# Get list of ports
+ports = config_db.get_table('PORT').keys()
+
+# Initialize the ports with zero initial packet drops
+drop_pkts = {port: 0 for port in ports}
+
+# Main handler function
+# Main handler function
+def handler():
+    """
+    Continuously monitors ports for dropped DHCP packets and logs them.
+    """
+    while True:
+        for port in drop_pkts.keys():
+            try:
+                output = subprocess.run(["tc", "-s", "qdisc", "show", "dev", str(port), "handle", "ffff:"], shell=False, capture_output=True)
+                if output.returncode == 0:  # Check for successful execution
+                    match = re.search(r'dropped (\d+)', output.stdout)
+                    if match:
+                        dropped_count = int(match.group(1))
+                        if dropped_count > drop_pkts[port]:
+                            logger.log_info(f"Port {port}: Current DHCP drop counter is {dropped_count}")
+                            drop_pkts[port] = dropped_count
+                        else:
+                            logger.log_warning(f"No new dropped packets found on port {port}")
```

Comment:
Why no dropped packets (Line 47 and line 49 log_warning) is treated as abnormal but dropped packets is treated as expected(line 44 log_info)? With this logic, whether actual DHCP packet rate low than tc rate treated as abnormal?

Reply:
Hi @yaqiangz , thanks for pointing that out. I have fixed this in the code. The systemd process will only log when dropped packets are observed, and it will do so as a warning, not as info.